contrib/src/main/java/org/apache/apex/malhar/contrib/kudu/AbstractKuduOutputOperator.java

```diff
@@ -0,0 +1,662 @@
+  {
+    ApexKuduConnection.ApexKuduConnectionBuilder apexKuduConnectionBuilder = getKuduConnectionConfig();
+    apexKuduConnection = apexKuduConnectionBuilder.build();
+    checkNotNull(apexKuduConnection,"Kudu connection cannot be null");
+    kuduTable = apexKuduConnection.getKuduTable();
+    kuduSession = apexKuduConnection.getKuduSession();
+    kuduClientHandle = apexKuduConnection.getKuduClient();
+    checkNotNull(kuduTable,"Kudu Table cannot be null");
+    checkNotNull(kuduSession, "Kudu session cannot be null");
+    allColumnDefs = new HashMap();
+    primaryKeyColumnNames = new HashSet<>();
+    kuduColumnBasedGetters = new HashMap();
+    buildGettersForPojoPayload();
+    reconcilingWindowId = Stateless.WINDOW_ID;
+    // The operator is working in a replay mode where the upstream buffer is re-streaming the tuples
+    // Note that there are two windows that need special core. The window that is being replayed and the subsequent
+    // window that might have resulted in a crash which we are referring as reconciling window
+    if ( (context.getValue(Context.OperatorContext.ACTIVATION_WINDOW_ID) != Stateless.WINDOW_ID) &&
+        context.getValue(Context.OperatorContext.ACTIVATION_WINDOW_ID) <
+        windowDataManager.getLargestCompletedWindow()) {
+      reconcilingWindowId = windowDataManager.getLargestCompletedWindow() + 1;
+    }
+
+    if ( (context.getValue(Context.OperatorContext.ACTIVATION_WINDOW_ID) != Stateless.WINDOW_ID) &&
+        context.getValue(Context.OperatorContext.ACTIVATION_WINDOW_ID) ==
+        windowDataManager.getLargestCompletedWindow()) {
+      reconcilingWindowId = windowDataManager.getLargestCompletedWindow();
+    }
+  }
+
+  @Override
+  public void deactivate()
+  {
+    try {
+      apexKuduConnection.close();
+    } catch (Exception e) {
+      LOG.error("Could not close kudu session and resources because " + e.getMessage(), e);
+    }
+  }
+
+  @Override
+  public void beforeCheckpoint(long l)
+  {
+    // Nothing to be done here. Child classes can use this if required
+  }
+
+  @Override
+  public void checkpointed(long l)
+  {
+    // Nothing to be done here. Child classes can use this if required
+  }
+
+  @Override
+  public void committed(long windowId)
+  {
+    try {
+      windowDataManager.committed(windowId);
+    } catch (IOException e) {
+      LOG.error("Error while committing the window id " + windowId + " because " + e.getMessage());
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    super.setup(context);
+    windowDataManager = getWindowDataManager();
+    if ( windowDataManager == null) {
+      windowDataManager = new FSWindowDataManager();
+    }
+    windowDataManager.setup(context);
+  }
+
+  @Override
+  public void teardown()
+  {
+    super.teardown();
+    windowDataManager.teardown();
+  }
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    super.beginWindow(windowId);
```

Comment:
My thinking is that it is a good practice to call the super methods especially when you are extending the base operator ( as the word base meant that it might have common logic and hence the call to super)

Removing this call as the base implementation does not have anything useful.